### ISSUE TITLE
Keras mismatches outputs and targets

### ISSUE THREAD

**Issue by georgwiese** [tensorflow/tensorflow#38313]
**System information** 
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS
- TensorFlow version (use command below): v2.0.0-rc2-26-g64c3d382ca 2.0.0
- Python version: 3.6.9

**Describe the current behavior**

- We build a keras model with outputs like `task_a_logits`, `task_a_probs`, `task_b_logits`, `task_b_probs`
- We compile our model by passing a `loss` dict like `{"task_a_logits": ..., "task_b_logits": ...}` (Note that no loss is applied on the probabilities). All losses are applied on the logits.
- We build a dataset that yields tuples like `(x, {"task_a_logits": ..., "task_b_logits": ...}`
- To `keras.fit()`, we pass a dataset that yields targets for each task, also in a dict.

This works without eager execution, but using TF 2.0's defaults, keras gets confused because there are less targets and losses than outputs. [This patch](https://gist.github.com/georgwiese/57568ac518813b9d9f0e6785d8f707fa) fixes the issue.

With the original code and the example above, `loss_fns` and `targets` have two elements and `outs` have four. They get zipped together and then shapes mismatch: `logits and labels must be broadcastable: logits_size=[18,4] labels_size=[18,3]`

**Describe the expected behavior**

Keras matches outputs and targets correctly.

**Standalone code to reproduce the issue** 
I provided a patch that points to the bug in the Keras source code.

**Comment by Saduf2019** [comment]
@georgwiese 
could you please provide with simple[complete] standalone code for us to replicate the issue

**Comment by Saduf2019** [comment]
@georgwiese
could you please update on the above comment

**Comment by Saduf2019** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
